src/lightning/pytorch/loggers/comet.py

```diff
@@ -241,21 +242,10 @@ def __init__(
         log.info(f"CometLogger will be initialized in {self.mode} mode")
 
         self._project_name: Optional[str] = project_name
-        self._experiment_key: Optional[str] = experiment_key
+        self._experiment_key: str = experiment_key or os.environ.get("COMET_EXPERIMENT_KEY") or comet_ml.generate_guid()
```

Comment:
That should be taken care of by the EM SDK already, no?

Reply:
I just tried to keep the same logger structure/behaviour and keep backward compatability - in previous version you could get experiment key by calling logger.version() even if real experiment is not yet created :)